app/components/provider_interface/dsa_success_page_component.rb

```diff
@@ -0,0 +1,42 @@
+module ProviderInterface
+  class DsaSuccessPageComponent < ViewComponent::Base
+    attr_reader :provider_user, :provider_permission_setup_pending
+
+    def initialize(provider_user:, provider_permission_setup_pending:)
```

Comment:
maybe if we pass in the feature flag value as a show new content field into this component it can simplify testing this/rendering the test component rather than using `instance_variable_set`?

Reply:
I kept it inside the component since that keeps the interface as simple as possible (and reduces the number of changes when deleting the flag). Since we're gong to remove it anyway, I think I'll leave it as is. Let me know if there's something I've missed though :)